ocaml/xapi/vm_platform.ml

```diff
@@ -145,16 +145,22 @@ let sanity_check ~platformdata ?firmware ~vcpu_max ~vcpu_at_startup ~domain_type
   end;
   if check_cores_per_socket && (List.mem_assoc "cores-per-socket" platformdata) then
     begin
+      let cps_str = List.assoc "cores-per-socket" platformdata in
+      let vcpus = Int64.to_int(vcpu_max) in
       try
-        let cores_per_socket = int_of_string(List.assoc "cores-per-socket" platformdata) in
+        let cps = int_of_string(cps_str) in
+
         (* VCPUs_max has to be a multiple of cores per socket *)
-        if ((Int64.to_int(vcpu_max) mod cores_per_socket) <> 0) then
-          raise (Api_errors.Server_error(Api_errors.invalid_value,
-                                         ["platform:cores-per-socket";
-                                          "VCPUs_max must be a multiple of this field"]))
+        if cps < 1 || ((vcpus mod cps) <> 0) then
+          raise (Api_errors.Server_error(
+                     Api_errors.invalid_value,
+                     [Printf.sprintf "platform:cores-per-socket (value %d)" cps;
+                      Printf.sprintf "VCPUs_max (value %d) must be a multiple of cores-per-socket" vcpus]))
       with Failure msg ->
```

Comment:
Seeing that code is guarding against negative values for `cps` I think it should do the same for 0. The modulo operation can fail with the exception `Division_by_zero` otherwise.

The exception can be caught using `with Failure _ | Division_by_zero ->` here

Reply:
`cps < 1` is true when `cps = 0`, and short-circuiting the `||` is part of the language specification: https://caml.inria.fr/pub/docs/manual-ocaml/expr.html#sec147

Reply:
fair enough